### PR TITLE
Chore/curate scripts and deps

### LINKED ISSUE #62: Make the package more lightweight
What I mean is that when ictinus is used in another project, whenever a new dependency is installed, ictinus reruns it's postinstall script which is huge.

*  make it so that the postinstall script is not run (either remove it completely or somehow configure it to not run when it's an npm package)
* slim down the postinstall script, if that's possible

### PR DESCRIPTION
**Pull request by mkarajohn**
Closes #62 

Reduced install time to 16.30s-21.75s. down from 147.27s.

This PR also cleans up the files that get downloaded on `npm install @orfium/ictinus`, by reworking the `.npmignore` file.

Now only the following will be downloaded on package install: 
image

Down from 
image

## Testing

You can test that everything works ok by installing `@orfium/ictinus@experimental`

### CODE REVIEW

**Comment by mkarajohn** [comment]
Also see this https://github.com/Orfium/orfium-ictinus/issues/65

**Comment by panvourtsis** [comment]
@mkarajohn there is also a need to change everywhere the
``` import ... from 'ictinus';```
to ```from '@orfium/ictinus';```

do you think we can add it now as this will be the release for npm usage that we didn't had till now

**Comment by mkarajohn** [comment]
@panvourtsis I have only left one unfixed eslint error, found in `utils/helpers.ts` because I am not exactly sure how I should fix it and retain the current functionality.

**Comment by mkarajohn** [comment]
I actually had to change all absolute imports to relative ones, because otherwise imports in the released package were not working properly, because they were still using absolute paths. The tsc does not bother with import paths.

Only thing I found that *could* possibly automate this conversion while allowing for use of absolute paths during development is [this](https://www.npmjs.com/package/tsconfig-paths-webpack-plugin) for which I am still not sure, too tired right now. Changing to relative paths seemed the easier option.

As a reference, Formik only seems to use relative imports too. (https://github.com/formium/formik/blob/master/packages/formik/src/Field.tsx)

**Comment by mkarajohn** [comment]
Easier to review this PR going by commits instead of files changed.

**Review by panvourtsis — COMMENTED** [review]
Why exactly ?

**Comment by mkarajohn** [comment]
> Why we need to change all files to relative paths ? i saw your commit but i dont understand

See https://github.com/Orfium/orfium-ictinus/pull/64#issuecomment-689880853

File: src/components/Button/Button.style.ts [hunk 1]
```diff
@@ -70,7 +69,10 @@ export const buttonSpanStyle = ({
   size,
 }: RequiredProperties<Props & { hasChildren: boolean }>) => (theme: Theme) => ({
   display: icon ? 'flex' : 'block',
-  flexDirection: (iconAlign === 'right' ? 'row-reverse' : 'row') as FlexDirectionProperty,
+  // In orfium-ictinus/node_modules/@emotion/serialize/node_modules/csstype/index.d.ts
+  // The FlexDirectionProperty, unlike other properties, does not include a simple 'string'
+  // definition. So we cast this as something it will accept.
+  flexDirection: iconAlign === 'right' ? ('row-reverse' as const) : ('row' as const),
```
[the remark this answers is not in the document]
Comment (mkarajohn) [reply src/components/Button/Button.style.ts:75]: I don't think that defining our own `FlexDirectionProperty` offers much advantage over casting it like that. We'll still have to change the type import in all places where it's used *and* now we have to "maintain" the flex direction definition ourselves, in which case at that point we may install `csstypes` ourselves and use `FlexDirectionProperty` from there directly. 

So personally, I would not bother with this. Feel free to make the change though.

**Comment by panvourtsis** [comment]
> @panvourtsis I have only left one unfixed eslint error, found in `utils/helpers.ts` because I am not exactly sure how I should fix it and retain the current functionality.

This as i saw it requires a bit of effort to fix, i suggest open a new issue for it and do it in another PR so we can merge this.

**Comment by mkarajohn** [comment]
All issues are resolved from my side, please approve or provide feedback, so this can be merged.

**Comment by michalisKout** [comment]
> All issues are resolved from my side, please approve or provide feedback, so this can be merged.

Nice! Please don't merge this to staging yet. I still need to fix some caching issues from previous builds in staging (Heroku CI).  I am working on it. I am sure you will face the same issues because this PR has changes in yarn.lock.

**Comment by mkarajohn** [comment]
@panvourtsis svg assets are now packages normally, please check if it works on VH when you have time

**Comment by panvourtsis** [comment]
> @panvourtsis svg assets are now packaged normally, please check if it works on VH when you have time

kudos, works correctly now

image

**Review by panvourtsis — APPROVED** [review]

**Comment by mkarajohn** [comment]
@panvourtsis please test again on VH. If it still does not work I quit my job

**Pull request merged**